bots/psuteam7bot/prophet.js

```diff
@@ -37,183 +55,171 @@ prophet.doAction = (self) => {
-                    return distance <= 16;
-                }
-            });
-
-            //general North/East/South/West guard post direction for checking for horizontal/ vertical movement
-            //N = 0
-            //E = 1
-            //S = 2
-            //W = 3
-            const compass = nearbyDefenders.length % 4;
-
-            //Get the direction value from directions array
-            let direction = movement.directions[compass*2];
-
-            let x = 1;  //movement amount for North East
-            let y = 3;
-
-            //If first four rotate left, otherwise rotate right
-            if(nearbyDefenders < 4)
-                direction = movement.rotateDirection(direction, -1);
-            else
-                direction = movement.rotateDirection(direction, 1);
-
-            x *= direction.x;
-            y *= direction.y;
-            self.log('x: ' + x + ' y: ' + y)
-
-            //Case for vertical movement
-            if(compass === 0 || compass === 2)
-                self.target = {x: self.base.x + x, y: self.base.y + y};
-            else
-                self.target = {x: self.base.x + y, y: self.base.y + x};
-
-            self.log("Assigned " + JSON.stringify(self.target) + " as guard post");
+            if(movement.aStarPathfinding(self, self.me, self.potentialEnemyCastleLocation[0], false)) {
+                self.log(self.path)
+            } else {
+                self.log('Cannot get path to guard post')
+                return;
+            }
         }
-        
-        //Movement to guard post should only take 5 turns (gave it a little more time to settle)
-        if(self.me.turn < 5)
-        {
-            const moveLocation = movement.moveTowards(self, self.target);
 
-            self.log("Moving towards guard post, targeting " + JSON.stringify(moveLocation));
-            return self.move(moveLocation.x-self.me.x, moveLocation.y-self.me.y);
-        }
+        self.log('DEFENDER prophet ' + self.id + ' moving towards guard post, Current: [' + self.me.x + ',' + self.me.y + ']')
+        return movement.moveAlongPath(self);
+    }
 
-        //Guarding behavior, doesn't flee, doesn't check fuel before attempting to attack
-        const attackable = combat.getAttackableEnemies(self);
+    //Guarding behavior, doesn't flee, doesn't check fuel before attempting to attack
+    const attackable = combat.getAttackableEnemies(self);
 
-        if(attackable.length > 0)
-        {
-            let attacking = attackable[0];
-            self.log("Attacking " + combat.UNITTYPE[attacking.unit] + " at " + attacking.x + ", " +  attacking.y);
-            return self.attack(attacking.x - self.me.x, attacking.y - self.me.y);
-        }
-        
-        //TODO store 'reclaimed' resources if capacity full and no enemies [future sprint obj]
-        //Otherwise, idle.
-        return;
+    if(attackable.length > 0)
+    {
```

Comment:
This is the code that we should prioritize first per my other comment. We may also tweak to account for visible enemies and make moves accordingly, but that's likely a Sprint 3 goal